src/main/java/com/nha/abdm/wrapper/hip/hrp/link/hipInitiated/HipLinkService.java

```diff
@@ -100,18 +100,20 @@ public FacadeResponse hipAuthInit(LinkRecordsRequest linkRecordsRequest) {
             .build();
 
     log.debug("LinkAuthInit : " + linkAuthInit.toString());
-
+    requestLogService.persistHipLinkRequest(linkRecordsRequest, RequestStatus.INITIATING, null);
     try {
       ResponseEntity<GenericResponse> response =
           requestManager.fetchResponseFromGateway(linkAuthInitPath, linkAuthInit);
       log.debug(linkAuthInitPath + " : linkAuthInit: " + response.getStatusCode());
       if (response.getStatusCode() == HttpStatus.ACCEPTED) {
-        requestLogService.persistHipLinkRequest(
```

Comment:
what was the reason you had to make all these changes?
Its good to explain the bug in the pull request description itself